### ISSUE TITLE
bigtable: emulator conditional filter bug

### ISSUE THREAD

**Issue by PaulMazzuca** [googleapis/google-cloud-go#1712]
## Client

BigTable Emulator with BigTable Python Library

## Describe Your Environment

```bah
# pip list | grep bigtable
google-cloud-bigtable    1.2.0 

# gcloud version
Google Cloud SDK 274.0.0
beta 2019.05.17
bigtable 
bq 2.0.51
cbt 
core 2019.12.17
docker-credential-gcr 
gsutil 4.46
kubectl 2019.11.04
```
## Expected Behavior

No results should be returned

## Actual Behavior

Results are returned

## Details

There is a bug in the big table emulator when a conditional filter is run using the python Bigtable library. This bug does not occur when running against the production Bigtable, so I am thinking that the issue is with the emulator.  If the chain does not return any results, then the conditional true filter should not return any results, but that is not the case when the code is run against the emulator.

```python
import os
from google.cloud import bigtable
import google.cloud.bigtable.row_filters as btf

os.environ["BIGTABLE_EMULATOR_HOST"] = "localhost:8086"
TEST_PROJECT_NAME = "test-project"
TEST_INSTANCE_NAME = "test-instance"

TEST_TABLE = "test-table"
TEST_FAMILY = "a"

admin_client = bigtable.Client(project=TEST_PROJECT_NAME, admin=True)
admin_inst = admin_client.instance(TEST_INSTANCE_NAME)
tbl = admin_inst.table(TEST_TABLE)
fam = tbl.column_family(TEST_FAMILY)
if not tbl.exists():
    tbl.create()
    fam.create()

# write a row
row = tbl.direct_row("aaa")
row.set_cell(TEST_FAMILY, b'a', '\x01')
row.set_cell(TEST_FAMILY, b'b', '\x05')
row.commit()

chain1 = btf.RowFilterChain([
    btf.ColumnQualifierRegexFilter(b'a'),
    btf.ValueRangeFilter(start_value='\x01')
    ])
chain2 = btf.RowFilterChain([
    btf.ColumnQualifierRegexFilter(b'b'),
    btf.ValueRangeFilter(start_value='\x09')
    ])

union_filt = btf.RowFilterUnion([chain1, chain2])

chain_filter = btf.RowFilterChain([
    union_filt,
    btf.CellsRowOffsetFilter(1)
    ])

cond_filter = btf.ConditionalRowFilter(
    chain_filter,
    true_filter=btf.PassAllFilter(True))

print("Just union")
for r in tbl.read_rows(filter_=union_filt):
    print(r.cells)

print("Just chain")
for r in tbl.read_rows(filter_=chain_filter):
    print(r.cells)

print("Cond Filter")
for r in tbl.read_rows(filter_=cond_filter):
    print(r.cells)

tbl.delete()
```

Results
```bash
Just union
{'a': {b'a': [<Cell value=b'\x01' timestamp=2019-12-27 19:23:27.402000+00:00>]}}
Just chain
Cond Filter. THIS SHOULD NOT RETURN ANYTHING
{'a': {b'a': [<Cell value=b'\x01' timestamp=2019-12-27 19:23:27.402000+00:00>], b'b': [<Cell value=b'\x05
' timestamp=2019-12-27 19:23:27.402000+00:00>]}}
```

**Comment by PaulMazzuca** [comment]
Thanks for taking a look at this. Any idea when it will be fixed/started?

**Comment by tbpg** [comment]
Thank you for filing this! I'll let the team that owns the emulator know, but no promises on the timeline, unfortunately.

**Comment by IlyaFaer** [comment]
@PaulMazzuca, thanks for the detailed report, I've reproduced the error (including the Python script). Looks like it's caused by empty rows, which are treated by the emulator like they are not empty. When the given row is filtered with the `chain_filter` within the emulator it looks like:
```go
&{a 0 [a b] map[a:[] b:[]]}
```
It becomes a `ReadRowsResponse` with `Chunks=nil` here:
https://github.com/googleapis/google-cloud-go/blob/23a470b5e634a30079cbb19b43fb3f05f1a861c8/bigtable/bttest/inmem.go#L446-L461

So in Python we're getting nothing ("Just chain" case).

But in the "Cond Filter" case, when `chain_filter` is used as a `PredicateFilter`, the same row is passed next to the `PassAllFilter`, and it's treated like a matched row. As `PredicateFilter` [is not actually applied](https://cloud.google.com/bigtable/docs/using-filters#condition) to the data, we're getting a matched row with all the original data.

Seems to me, the fix is to check the row while filtering: if it contains no cells then unmatch it (return `false` in the first result). I've already tried to raw-code it:

![Безымянный](https://user-images.githubusercontent.com/26030071/90522543-827e6980-e174-11ea-802f-a37252252607.png)

It fixes this issue bug. No tests failed with it, so I assume we can do this (and it should be done for all the emulator filters, as I understood).

/cc @tritone
